citellusclient/plugins/core/network/interface-onboot.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# Copyright (C) 2020 Volkan Yalcin <vlyalcin@gmail.com>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# long_name: Reports interfaces that its state 'up' and doesn't have ONBOOT='yes' in its network-scripts file.
+# description: Reports interfaces that its state 'up' and doesn't have ONBOOT='yes' in its network-scripts file.
+# bugzilla: https://github.com/citellusorg/citellus/issues/535
+# priority: 100
+# kb:
+
+# Test cases
+# 1 Give an invalid SOS_IP_ADDRESS_PATHS. Observe the fail.
+# 2 Remove ONBOOT=yes from network file for the UP interfaces. Observe the fail.
+
+# Load common functions
+[[ -f "${CITELLUS_BASE}/common-functions.sh" ]] && . "${CITELLUS_BASE}/common-functions.sh"
+
+NETWORK_SCRIPTS_PATH="/etc/sysconfig/network-scripts/ifcfg"
+SOS_IP_ADDRESS_PATHS="sos_commands/networking/ip_-d_address sos_commands/networking/ip_address"
+
+if [[ ${CITELLUS_LIVE} -eq 0 ]]; then
+    NETWORK_SCRIPTS_PATH="${CITELLUS_ROOT}${NETWORK_SCRIPTS_PATH}"
+    IP_ADDRESS_FILE=""
+    for SOS_IP_ADDRESS_PATH in ${SOS_IP_ADDRESS_PATHS}; do
+        if [ -f "${CITELLUS_ROOT}/${SOS_IP_ADDRESS_PATH}" ]; then
+            IP_ADDRESS_FILE="${CITELLUS_ROOT}/${SOS_IP_ADDRESS_PATH}"
+            break
+        fi
+    done
+    if [ -z "$IP_ADDRESS_FILE" ]; then
```

Comment:
you can use the macro (from common-functions) `is_required_file` as it will automatically exit and report the missing file so test exits

Reply:
It can be combined with `first_file_available` to pass list of files so that it returns the first one available, like the way it's done on `clock-1-ntpd.sh`:

~~~sh
    FILE=$(first_file_available "${CITELLUS_ROOT}/sos_commands/ntp/ntpq_-p" "${CITELLUS_ROOT}/sos_commands/ntp/ntpq_-pn")

    is_required_file "${FILE}"
~~~

Reply:
first_file_available is pretty good. reduced the line of the code.